src/array_decoder/timestamp.rs

```diff
@@ -103,7 +103,7 @@ fn decimal128_decoder(
     let data = get_rle_reader(column, data)?;
 
     let secondary = stripe.stream_map().get(column, Kind::Secondary);
-    let secondary = get_rle_reader(column, secondary)?;
+    let secondary = get_unsigned_rle_reader(column, secondary);
```

Comment:
Bug was here. Secondary is unsigned stream. It's retrieved correctly for timestamp, but was mistakenly taken as signed here. See line 62 for correct behaviour.

https://github.com/datafusion-contrib/orc-rust/blob/f934c8eab4915836b8655e61bec5ddd959113c50/src/array_decoder/timestamp.rs#L53-L72